packages/opentelemetry-exporter-collector/src/platform/browser/CollectorTraceExporter.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CollectorTraceExporterBase } from '../../CollectorTraceExporterBase';
+import { ReadableSpan } from '@opentelemetry/tracing';
+import { toCollectorExportTraceServiceRequest } from '../../transform';
+import * as collectorTypes from '../../types';
+import { CollectorExporterConfigBrowser } from '../../types';
+import { sendWithBeacon, sendWithXhr } from './util';
+
+const DEFAULT_COLLECTOR_URL = 'http://localhost:55678/v1/trace';
+
+/**
+ * Collector Exporter for Web
+ */
+export class CollectorTraceExporter extends CollectorTraceExporterBase<
+  CollectorExporterConfigBrowser
+> {
+  DEFAULT_HEADERS: { [key: string]: string } = {
+    [collectorTypes.OT_REQUEST_HEADER]: '1',
+  };
+  private _headers: { [key: string]: string };
+  private _useXHR: boolean = false;
+
+  /**
+   * @param config
+   */
+  constructor(config: CollectorExporterConfigBrowser = {}) {
+    super(config);
+    this._headers = config.headers || this.DEFAULT_HEADERS;
+    this._useXHR =
+      !!config.headers || typeof navigator.sendBeacon !== 'function';
+  }
+
+  onInit(): void {
+    window.addEventListener('unload', this.shutdown);
+  }
+
+  onShutdown(): void {
+    window.removeEventListener('unload', this.shutdown);
+  }
```

Comment:
This all seems exactly the same as the CollectorMetricExporter. Should these classes share a common ancestor abstract class?

Reply:
Yeah I was thinking of doing that too, the problem I was running into was that the CollectorMetricExporter inherits from the MetricExporter and the CollectorTraceExporter inherits from the SpanExporter.

This is important, because this allows them to be inputted into the TraceProvider and MetricProvider